### PR TITLE
Update advanced models to AllenNLP 1+

### PR DESCRIPTION
**Pull request by GillesJ**
I updated the config and model code for the advanced models.
I know these aren't currently discussed in the tutorial, but I found the code helpful.

What changed:
- configs: replaced `iterator` with `dataloader`, wrapped `embedders` with `token_embedders` dict as required by AllenNLP v1+.
- model code for `lstm_character.py`: init output dict and place logits in there for the predictor.

Tested and working with AllenNLP v1.1.0.

### CODE REVIEW

**Comment by jbarrow** [comment]
Thanks for the PR! Going to test these out on my local machine and then merge!
